### PR TITLE
Improve YAML parsing to handle list sequences within SLI spec

### PR DESCRIPTION
**Pull request by aaranmcguire**
This PR will allow more YAML configurations within the SLI spec as lists were previously not parsed.

Within my org we're wanting to allow our users to see SLOs by cluster, or server. This is likely niche, but we're doing this by allowing users to define the Prometheus labels to scrape. e.g. `sum(http_requests_total{}) by (cluster)`. To do this we're wanting the spec to look like the following.

```yaml
indicator:
    spec:
      ratioMetric:
        total:
          metricSource:
            type: Prometheus
            metricSourceRef: thanos
            spec:
              query: http_requests_total{service="example"}
              dimensions:
                - cluster
                - server
```

This may be niche, but we believe that the spec was written to be open minded regarding the format within the `metricSource.spec`, so this should be supported.

FYI, the output of this within the Go struct matches the existing format with `dimensions` being `cluster;server`

### CODE REVIEW

**Comment by nieomylnieja** [comment]
hey @aaranmcguire, thanks for the contribution!
this looks good, but linter reports some errors, please correct them :)

you might be interested in https://github.com/OpenSLO/OpenSLO/pull/277, as this `oslo` code will be replaced by the `openslo` package. Would you mind contributing to the branch of my PR there as well?

File: pkg/manifest/v1/indicator.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type SLI struct {
 
 // SLIInline represents the SLI inline.
 type SLIInline struct {
-	Metadata Metadata `yaml:"metadata" validate:"required"`
+	Metadata Metadata `yaml:"metadata,omitempty" validate:"omitempty"`
```
Comment (nieomylnieja) [inline pkg/manifest/v1/indicator.go:43]: is this intentional change? If so, please revert this, `metadata` is a required field for inline definitions

File: pkg/manifest/v1/alerts.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ type AlertCondition struct {
 // different from the AlertCondition type because it does not have an APIVersion.
 type AlertConditionInline struct {
 	Kind     string             `yaml:"kind" validate:"required"`
-	Metadata Metadata           `yaml:"metadata" validate:"required"`
+	Metadata Metadata           `yaml:"metadata,omitempty" validate:"omitempty"`
```
Comment (nieomylnieja) [inline pkg/manifest/v1/alerts.go:29]: is this intentional change? If so, please revert this, `metadata` is a required field for inline definitions

**Review by nieomylnieja — CHANGES_REQUESTED** [review]

File: pkg/manifest/v1/alerts.go (hunk 2 above)
Comment (aaranmcguire) [reply pkg/manifest/v1/alerts.go:29]: My bad. I was experimenting with another change. Will revert now.

**Comment by aaranmcguire** [comment]
I've really F'd this one up with the force push, let me just do it again from scratch.

**Comment by nieomylnieja** [comment]
On seconds thoughts and inspection of the current, `v1` schema description, I think `oslo` has at a wrong treating `spec` as a `map[string]string`, the OpenSLO specification states:

>arbitrary chosen fields for every data source type to make it comfortable to use
anything that is valid YAML can be put here.

This means that the SLI `spec` should in fact be `map[string]interface{}`.

I've added these changes in the following OpenSLO commit:
https://github.com/OpenSLO/OpenSLO/pull/276/commits/c8354c5aac3b5ea4eade1ce4ad473d084e257902

**Pull request closed without merging**